types/Address/Address.d.ts

```diff
@@ -0,0 +1,142 @@
+import { IObjectWithId } from '../base';
+import { DeepPartial } from '../utils';
+import { IAddressCreateParameters } from './AddressCreateParameters';
+import { IVerifications } from './Verifications';
+
+/**
+ * Address objects are used to represent people, places, and organizations in a number of contexts.
+ * For example, a Shipment requires a to_address and from_address to accurately calculate rates and generate postage.
+ *
+ * Additionally, EasyPost offers several verification tools that can be used to detect deliverability issues, correct minor errors in spelling/formatting, and determine if an Address is residential or not (which has a significant effect on Shipment rating for many carriers).
+ *
+ * @see https://www.easypost.com/docs/api/node#address-object
+ */
+export declare interface IAddress extends IObjectWithId<'Address'> {
+  /**
+   * First line of the address
+   */
+  street1: string;
+
+  /**
+   * Second line of the address
+   */
+  street2: string;
+
+  /**
+   * City the address is located in
+   */
+  city: string;
+
+  /**
+   * State or province the address is located in
+   */
+  state: string;
+
+  /**
+   * ZIP or postal code the address is located in
+   */
+  zip: string;
+
+  /**
+   * ISO 3166 country code for the country the address is located in
+   */
+  country: string;
+
+  /**
+   * Whether or not this address would be considered residential
+   */
+  residential: boolean;
+
+  /**
+   * The specific designation for the address (only relevant if the address is a carrier facility)
+   */
+  carrier_facility: string;
+
+  /**
+   * Name of the person. Both name and company can be included
+   */
+  name: string;
+
+  /**
+   * Name of the organization. Both name and company can be included
+   */
+  company: string;
+
+  /**
+   * Phone number to reach the person or organization
+   */
+  phone: string;
+
+  /**
+   * Email to reach the person or organization
+   */
+  email: string;
+
+  /**
+   * Federal tax identifier of the person or organization
+   */
+  federal_tax_id: string;
+
+  /**
+   * State tax identifier of the person or organization
+   */
+  state_tax_id: string;
+
+  /**
+   * The result of any verifications requested
+   */
+  verifications: IVerifications;
+}
+
+export declare class Address implements IAddress {
+  public constructor(input: DeepPartial<IAddressCreateParameters>);
+
+  street1: string;
+  street2: string;
```

Comment:
What about some of these properties that are optional such as `street2`, I'm not seeing question marks or other defining syntax that would indicate these are optional? Would this then require that all these properties be present when creating an address?

Reply:
You are right here. Some background, I copied these types last year when I was building an EP implementation for someone, and I've had to make quite a few modifications. I wish I had enumerated everything I've done, but I didn't. I can say that this is what I have for the Address class properties:
```typescript
    street1: string | null;
    street2: string | null;
    city: string | null;
    state: string | null;
    zip: string;
    country: string;
    residential: boolean | null;
    carrier_facility: string | null;
    name: string | null;
    company: string | null;
    phone: string | null;
    email: string | null;
    federal_tax_id: string | null;
    state_tax_id: string | null;
    verifications: IVerifications;
    id: string;
    mode: 'test' | 'production';
    object: 'Address';
```

and my `IAddress` interface:
```typescript
export interface IAddress extends IObjectWithId<'Address'> {
    /**
     * First line of the address
     */
    street1: string | null;

    /**
     * Second line of the address
     */
    street2: string | null;

    /**
     * City the address is located in
     */
    city: string | null;

    /**
     * State or province the address is located in
     */
    state: string | null;

    /**
     * ZIP or postal code the address is located in
     */
    zip: string | null;

    /**
     * ISO 3166 country code for the country the address is located in
     */
    country: string;

    /**
     * Whether or not this address would be considered residential
     */
    residential: boolean | null;

    /**
     * The specific designation for the address (only relevant if the address is a carrier facility)
     */
    carrier_facility: string | null;

    /**
     * Name of the person. Both name and company can be included
     */
    name: string | null;

    /**
     * Name of the organization. Both name and company can be included
     */
    company: string | null;

    /**
     * Phone number to reach the person or organization
     */
    phone: string | null;

    /**
     * Email to reach the person or organization
     */
    email: string | null;

    /**
     * Federal tax identifier of the person or organization
     */
    federal_tax_id: string | null;

    /**
     * State tax identifier of the person or organization
     */
    state_tax_id: string | null;

    /**
     * The result of any verifications requested
     */
    verifications: IVerifications | null;
  }
```